### PR TITLE
refactored to use existing models 

### PR DESCRIPTION
**Pull request by ghost**
All the tests with the code I touched passes, I will work on fixing all the existing ones that currently break.

### CODE REVIEW

File: e2e-disabled/AccountListener.spec.ts [hunk 1]
```diff
@@ -22,16 +22,16 @@
...
-import {Account} from "../../src/models/account/Account";
-import {Address} from "../../src/models/account/Address";
-import {XEM} from "../../src/models/mosaic/XEM";
-import {NetworkTypes} from "../../src/models/node/NetworkTypes";
-import {EmptyMessage} from "../../src/models/transaction/PlainMessage";
-import {TimeWindow} from "../../src/models/transaction/TimeWindow";
-import {TransferTransaction} from "../../src/models/transaction/TransferTransaction";
-import {NEMLibrary} from "../../src/NEMLibrary";
+import {AccountListener} from "../src/infrastructure/AccountListener";
+import {TransactionHttp} from "../src/infrastructure/TransactionHttp";
```
Comment (spentak) [inline e2e-disabled/AccountListener.spec.ts:26]: add spaces on this imports curly braces

File: e2e-disabled/AccountListener.spec.ts [hunk 2]
```diff
@@ -22,16 +22,16 @@
...
+import {Account} from "../src/models/account/Account";
+import {Address} from "../src/models/account/Address";
+import {XEM} from "../src/models/mosaic/XEM";
+import {NetworkTypes} from "../src/models/node/NetworkTypes";
+import {EmptyMessage} from "../src/models/transaction/PlainMessage";
+import {TimeWindow} from "../src/models/transaction/TimeWindow";
+import { TransferTransaction, TxType } from "../src/models/transaction/TransferTransaction";
```
Comment (spentak) [inline e2e-disabled/AccountListener.spec.ts:33]: Let's probably be more verbose on this type TxType - maybe TransactionType or whatever it is

File: e2e-disabled/TransactionHttp.spec.ts [hunk 3]
```diff
@@ -22,47 +22,44 @@
...
+import { TimeWindow } from "../src/models/transaction/TimeWindow";
+import { TransferTransaction, TxType } from "../src/models/transaction/TransferTransaction";
+import { NEMLibrary } from "../src/NEMLibrary";
+import { TestVariables } from "../test/config/TestVariables.spec";
 
 declare let process: any;
 
 describe("TransactionHttp", () => {
-  const recipientAccount: string = "TBV7LE4TFDEMGVOON5MYOK2P7TU2KEKLMHOLHQT6";
+  const recipientAccount: string = "TDU225EF2XRJTDXJZOWPNPKE3K4NYR277EQPOPZD";
```
Comment (spentak) [inline e2e-disabled/TransactionHttp.spec.ts:61]: Is the type declaration necessary?

File: e2e-disabled/TransactionHttp.spec.ts [hunk 4]
```diff
@@ -122,10 +119,10 @@ describe("TransactionHttp", () => {
...
-    const transferTransaction = TransferTransaction.createWithMosaics(
-      TimeWindow.createWithDeadline(),
+    const transferTransaction = TransferTransaction.create(
       new Address(recipientAccount),
-      [new XEM(10)],
+      TxType.other,
+      new XEM(10),
       EmptyMessage);
     const signedTransaction = account.signTransaction(transferTransaction);
     // transactionHttp.announceTransaction(signedTransaction).subscribe(announceSuccessResult => {
```
Comment (spentak) [inline e2e-disabled/TransactionHttp.spec.ts:128]: i'm hesistant to publish any commented code in the public repo

File: e2e-disabled/TransferTransactionTestnetFees.spec.ts [hunk 5]
```diff
@@ -22,22 +22,21 @@
...
+import { EmptyMessage, PlainMessage } from "../src/models/transaction/PlainMessage";
+import { TransferTransaction, TxType } from "../src/models/transaction/TransferTransaction";
+import { NEMLibrary } from "../src/NEMLibrary";
+import { TestVariables } from "../test/config/TestVariables.spec";
 
 declare let process: any;
 
 describe("TransferTransactionTestnetFees", () => {
-  const recipientAccount: string = "TBV7LE4TFDEMGVOON5MYOK2P7TU2KEKLMHOLHQT6";
+  const recipientAccount: string = "TDU225EF2XRJTDXJZOWPNPKE3K4NYR277EQPOPZD";
```
Comment (spentak) [inline e2e-disabled/TransferTransactionTestnetFees.spec.ts:39]: Explicit typing probably not necessary

File: e2e/infrastructure/AddressListener.spec.ts [hunk 6]
```diff
@@ -23,31 +23,28 @@
...
 declare let process: any;
 
 describe("ConfirmedTransactionListener", () => {
   const privateKey: string = process.env.PRIVATE_KEY;
   let transactionHttp: TransactionHttp;
-  let account: CacheAccount;
+  let account: Account;
 
   before(() => {
     // Initialize NEMLibrary for TEST_NET Network
```
Comment (spentak) [inline e2e/infrastructure/AddressListener.spec.ts:45]: haha a pointless comment

File: src/models/account/Address.ts [hunk 7]
```diff
@@ -22,7 +22,12 @@
  * SOFTWARE.
  */
 
+import { Observable } from 'rxjs/Observable';
+import { mapTransfer, transferFilter } from '../../utilities/TransactionUtilities';
+import { ConfirmedTransactionListener } from '../../infrastructure/ConfirmedTransactionListener';
+import { nodeEndpoints } from '../../utilities/NodeEndpointUtilities';
 import {NetworkTypes} from "../node/NetworkTypes";
```
Comment (spentak) [inline src/models/account/Address.ts:29]: fix spacing

File: src/models/account/Address.ts [hunk 8]
```diff
@@ -69,4 +74,12 @@ export class Address {
   public equals(otherAddress: Address) {
     return this.plain() == otherAddress.plain();
   }
+
+  /**
+   * Start listening new confirmed cache transactions
```
Comment (spentak) [inline src/models/account/Address.ts:79]: Can we create a better description for this comment

Also this public confirmedTxObserver - is this something an developer is going to call? Does it need to be exposed.

File: src/models/transaction/TransferTransaction.ts [hunk 9]
```diff
@@ -137,6 +152,40 @@ export class TransferTransaction extends Transaction {
     return this._mosaics!.map((_) => _.mosaicId);
   }
 
+  /**
+   * returns mosaic details without sorting through array
+   * @returns MosaicTransferable
+   */
+  public mosaicDetails = (): XEM | CACHE | MosaicTransferable => {
```
Comment (spentak) [inline src/models/transaction/TransferTransaction.ts:159]: Does XEM and CACHE inherit from MosaicTransferable? If so then we could just use polymorphism and do:
`public mosaicDetails = (): MosaicTransferable`

File: src/models/transaction/TransferTransaction.ts [hunk 10]
```diff
@@ -137,6 +152,40 @@ export class TransferTransaction extends Transaction {
...
+      return mosaicDetails(this);
+    } else {
+      return xemDetails(this);
+    }
+  };
+
+  /**
+   * Create a CacheTransferTransaction object
+   * @param txType - TxType.xem, .cache, .other
+   * @param recipient
```
Comment (spentak) [inline src/models/transaction/TransferTransaction.ts:170]: finish adding comments. Also again i think we should be verbose whereever possible: txType maybe instead transactionType

File: src/models/transaction/TransferTransaction.ts [hunk 11]
```diff
@@ -160,22 +209,22 @@ export class TransferTransaction extends Transaction {
 
   /**
    * Create a TransferTransaction object
-   * @param timeWindow
    * @param recipient
    * @param xem
```
Comment (spentak) [inline src/models/transaction/TransferTransaction.ts:213]: Add comments for params

File: src/models/transaction/TransferTransaction.ts [hunk 12]
```diff
@@ -190,16 +239,16 @@ export class TransferTransaction extends Transaction {
 
   /**
    * Create a TransferTransaction object
-   * @param timeWindow
    * @param recipient
```
Comment (spentak) [inline src/models/transaction/TransferTransaction.ts:242]: comments for params

File: src/services/QRService.ts [hunk 13]
```diff
@@ -171,8 +170,8 @@ export class QRService {
    */
   public decryptTrasactionQRText(qrTransactionText: QRTransactionText): TransferTransaction {
```
Comment (spentak) [inline src/services/QRService.ts:171]: So this function says "decrypt" but it looks like its doing a lot more than decrypting. And i'm not sure i see where it actually IS decrypting anything.

File: src/utilities/NodeEndpointUtilities.ts [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import { WebSocketConfig } from '../infrastructure/Listener';
+import { NetworkTypes } from '../models/node/NetworkTypes';
+import { NEMLibrary } from '../NEMLibrary';
+
+export const nodeEndpoints = (): Array<WebSocketConfig> => {
+  if (NEMLibrary.getNetworkType() === NetworkTypes.MAIN_NET) {
```
Comment (spentak) [inline src/utilities/NodeEndpointUtilities.ts:30]: I'm not sure i understand why this function was created? What does this do that the code didn't have before?

File: src/utilities/TransactionUtilities.ts [hunk 15]
```diff
@@ -85,13 +86,15 @@ export const mapTransfer = (transaction: Transaction): CacheTransferTransaction
...
  * @returns {boolean}
  */
-export const cacheDetails = (transaction: TransferTransaction): CACHE => {
-  if (transaction.containsMosaics()) {
-    const cache = transaction.mosaics().find(mosaic => mosaic.mosaicId.namespaceId === CACHE.MOSAICID.namespaceId && mosaic.mosaicId.name === CACHE.MOSAICID.name);
-    if (cache) { return new CACHE(cache.quantity / Math.pow(10, CACHE.DIVISIBILITY)); }
-  }
-  return new CACHE(0);
-};
+export const mosaicDetails = (transaction: TransferTransaction): MosaicTransferable | CACHE => {
```
Comment (spentak) [inline src/utilities/TransactionUtilities.ts:89]: So i dont understand why this function and the others are in a file called TransactionUtilities? Is this the best place for them? Wouldn't it be better to Mosaic.mosaicDetails() or something?

File: src/utilities/TransactionUtilities.ts [hunk 16]
```diff
@@ -85,13 +86,15 @@ export const mapTransfer = (transaction: Transaction): CacheTransferTransaction
...
+  const mosaics = transaction.mosaics();
+    if (mosaics[0].mosaicId.namespaceId === CACHE.MOSAICID.namespaceId && mosaics[0].mosaicId.name === CACHE.MOSAICID.name) {
```
Comment (spentak) [inline src/utilities/TransactionUtilities.ts:91]: seems dangerous to access index 0

File: src/utilities/TransactionUtilities.ts [hunk 17]
```diff
@@ -85,13 +86,15 @@ export const mapTransfer = (transaction: Transaction): CacheTransferTransaction
...
+      return new CACHE(mosaics[0].quantity / Math.pow(10, CACHE.DIVISIBILITY));
```
Comment (spentak) [inline src/utilities/TransactionUtilities.ts:92]: We still keeping CACHE ?

File: e2e-disabled/TransactionHttp.spec.ts (hunk 4 above)
Comment (ghost) [reply e2e-disabled/TransactionHttp.spec.ts:128]: This is commented from their tests, haven't touched it yet

File: e2e/infrastructure/AddressListener.spec.ts (hunk 6 above)
Comment (ghost) [reply e2e/infrastructure/AddressListener.spec.ts:45]: again their code

File: src/models/account/Address.ts (hunk 8 above)
Comment (ghost) [reply src/models/account/Address.ts:79]: yes, this is the listener based off the address. Address.addObserver it used to be.

File: src/services/QRService.ts (hunk 13 above)
Comment (ghost) [reply src/services/QRService.ts:171]: their code

File: src/utilities/NodeEndpointUtilities.ts (hunk 14 above)
Comment (ghost) [reply src/utilities/NodeEndpointUtilities.ts:30]: i was requiring this to be passed in to connect to a node, there are issues automatically connecting to the list of nodes, its def temporary.  Remember you asked why you have to pass this in to the observer, this is that

File: src/utilities/TransactionUtilities.ts (hunk 17 above)
Comment (ghost) [reply src/utilities/TransactionUtilities.ts:92]: yeah, why wouldn't we have our token embedded in the sdk?

**Pull request closed without merging**